src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -6704,6 +6720,705 @@ namespace HeatBalanceSurfaceManager {
+                        // currently set to mean air temp but should add error warning here
+                        RefAirTemp(surfNum) = MAT(zoneNum);
+                        TempEffBulkAir(surfNum) = MAT(zoneNum); // for reporting surf adjacent air temp
+                    }
+                }
+            }
+
+            // Following variables must be reset due to possible recall of this routine by radiant and Resimulate routines.
+            // CalcWindowHeatBalance is called, then, multiple times and these need to be initialized before each call to
+            // CalcWindowHeatBalance.
+            // Only for Surface(SurfNum).Class == DataSurfaces::SurfaceClass_Window
+            int const firstWindowSurf = Zone(zoneNum).WindowSurfaceFirst;
+            int const lastWindowSurf = Zone(zoneNum).WindowSurfaceLast;
+            for (int surfNum = firstWindowSurf; surfNum <= lastWindowSurf; ++surfNum) {
+                WinHeatGain(surfNum) = 0.0;
+                WinHeatTransfer(surfNum) = 0.0;
+                WinHeatGainRep(surfNum) = 0.0;
+                WinHeatLossRep(surfNum) = 0.0;
+                WinGainConvGlazToZoneRep(surfNum) = 0.0;
+                WinGainIRGlazToZoneRep(surfNum) = 0.0;
+                WinLossSWZoneToOutWinRep(surfNum) = 0.0;
+                WinGainFrameDividerToZoneRep(surfNum) = 0.0;
+                WinGainConvGlazShadGapToZoneRep(surfNum) = 0.0;
+                WinGainConvShadeToZoneRep(surfNum) = 0.0;
+                OtherConvGainInsideFaceToZoneRep(surfNum) = 0.0;
+                WinGainIRShadeToZoneRep(surfNum) = 0.0;
+                SurfaceWindow(surfNum).FrameQRadOutAbs = 0.0;
+                SurfaceWindow(surfNum).FrameQRadInAbs = 0.0;
+                SurfaceWindow(surfNum).DividerQRadOutAbs = 0.0;
+                SurfaceWindow(surfNum).DividerQRadInAbs = 0.0;
+            }
+
+        // Calculate heat extract due to additional heat flux source term as the surface boundary condition - all HT surfaces
+            if (DataSurfaces::AnyHeatBalanceInsideSourceTerm) {
+                for (int surfNum = firstSurf; surfNum <= lastSurf; ++surfNum) {
+                    if (Surface(surfNum).InsideHeatSourceTermSchedule) {
+                        QAdditionalHeatSourceInside(surfNum) =
+                            EnergyPlus::ScheduleManager::GetCurrentScheduleValue(Surface(surfNum).InsideHeatSourceTermSchedule);
+                    }
+                }
+            }
+
+            // Set up coefficient arrays prior to calculations and precalc terms that do no change during iteration - non-window surfaces
+            int const firstNonWinSurf = Zone(zoneNum).NonWindowSurfaceFirst;
+            int const lastNonWinSurf = Zone(zoneNum).NonWindowSurfaceLast;
+            Real64 const timeStepZoneSeconds = TimeStepZoneSec; // local for vectorization
+            // this loop auto-vectorizes
+            for (int surfNum = firstNonWinSurf; surfNum <= lastNonWinSurf; ++surfNum) {
```

Comment:
Woohoo! Or more like whoopdeedoo, because it's not a huge performance gain, but it's a starting point. Two other loops below also auto-vectorize.